### PR TITLE
[Ruby 2.7] Add new specs. Language/Core

### PR DESCRIPTION
**Pull request by andrykonchin**
### Language

- Setting $; to a non-nil value is warned now. [Feature #14240]
- Setting $, to a non-nil value is warned now. [Feature #14240]
- Modifier rescue now operates the same for multiple assignment as single assignment. [Bug #8279]

### Core

- Added Array#intersection. [Feature #16155]
- Added Enumerator.produce to generate an Enumerator from any custom
data transformation. [Feature #14781]
- Added Enumerator::Lazy#eager that generates a non-lazy enumerator
from a lazy enumerator. [Feature #15901]
- Added Enumerator::Yielder#to_proc so that a Yielder object can be directly passed to another method as a block argument. [Feature #15618]
- Integer#[] now supports range operations. Feature #8842
- Time#inspect is separated from Time#to_s and it shows
the time's sub second. Feature #15958

### CODE REVIEW

**Comment by andrykonchin** [comment]
@eregon Looks like the specs aren't run against Ruby 2.7. Naive adding 2.7 to the versions list in `.github/workflows/ci.yml` doesn't help.

So is it known how to set up 2.7?

**Comment by eregon** [comment]
@andrykonchin That's unfortunate, it seems https://github.com/actions/setup-ruby does not support Ruby 2.7 yet.

Maybe we should add TravisCI back until it does?

**Comment by eregon** [comment]
I filed https://github.com/actions/setup-ruby/issues/45, but I guess it won't be fixed so soon.

**Comment by eregon** [comment]
I added 2.7 in TravisCI, just on Linux currently, until we figure out how to run 2.7 in Actions: https://github.com/ruby/spec/pull/746

**Comment by andrykonchin** [comment]
@eregon Great. Thanks.

File: core/time/inspect_spec.rb [hunk 1]
```diff
@@ -3,4 +3,19 @@
...
+      t = Time.utc(2007, 11, 1, 15, 25, 0, 123456)
+      t.inspect.should == "2007-11-01 15:25:00.123456 UTC"
+    end
+
+    it "formats nanoseconds as a Rational" do
+      t = Time.utc(2007, 11, 1, 15, 25, 0, 123456.789)
+      t.nsec.should == 123456789
+      t.strftime("%N").should == "123456789"
+
+      t.inspect.should == "2007-11-01 15:25:00 8483885939586761/68719476736000000 UTC"
```
Comment (andrykonchin) [inline core/time/inspect_spec.rb]: Created an issue https://bugs.ruby-lang.org/issues/16470 for this case

File: language/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    it "can be placed between fluent dot now" do
+      code = <<~CODE
+        10
+          # some comment
+          .to_s
+        CODE
+
+      -> do
+        eval(code).should == '10'
+      end.should_not raise_error
```
[the remark this answers is not in the document]
Comment (andrykonchin) [reply language/comment_spec.rb]: @eregon Done. Thank you

**Pull request merged**